src/components/PinchZoom/PinchZoom.tsx

```diff
@@ -43,83 +43,82 @@ export interface PinchZoomProps extends React.HTMLAttributes<HTMLDivElement> {
+      (zoomRef: ReactZoomPanPinchRef) => {
+        const { state, instance } = zoomRef;
+        if (state.scale > 0.99 && state.scale < 1.01) {
+          setIsZoomed(false);
+          onZoomChange?.(false);
+        } else {
+          setIsZoomed(true);
+          onZoomChange?.(true);
+        }
 
-      const isAtLeftEdge = Math.abs(state.positionX - (instance.bounds?.maxPositionX ?? 0)) < 0.01;
-      const isAtRightEdge = Math.abs(state.positionX - (instance.bounds?.minPositionX ?? 0)) < 0.01;
+        const isAtLeftEdge = Math.abs(state.positionX - (instance.bounds?.maxPositionX ?? 0)) < 0.01;
+        const isAtRightEdge = Math.abs(state.positionX - (instance.bounds?.minPositionX ?? 0)) < 0.01;
 
-      onAtLeftEdge?.(isAtLeftEdge);
-      onAtRightEdge?.(isAtRightEdge);
-    },
-    [onZoomChange, onAtLeftEdge, onAtRightEdge],
-  );
+        onAtLeftEdge?.(isAtLeftEdge);
+        onAtRightEdge?.(isAtRightEdge);
+      },
+      [onZoomChange, onAtLeftEdge, onAtRightEdge],
+    );
 
-  return (
-    <TransformWrapper
-      maxScale={maxZoom}
-      onTransformed={onTransformed}
-      minScale={1}
-      disablePadding
-      doubleClick={{
-        mode: isZoomed ? 'reset' : 'toggle',
-      }}
-      wheel={{
-        smoothStep: 0.01,
-      }}
-      panning={{
-        disabled: !isZoomed,
-      }}
-      ref={zoomRef}
-    >
-      <TransformComponent
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore
-        wrapperProps={{ ...commonProps, ...props }}
-        wrapperClass={classNames(baseClassName, className, {
-          [`${baseClassName}-zoomed`]: isZoomed,
-        })}
-        contentClass={`${baseClassName}-content`}
+    return (
+      <TransformWrapper
+        maxScale={maxZoom}
+        onTransformed={onTransformed}
+        minScale={1}
+        disablePadding
+        doubleClick={{
+          mode: isZoomed ? 'reset' : 'toggle',
+        }}
+        wheel={{
+          smoothStep: 0.01,
+        }}
+        panning={{
+          disabled: !isZoomed,
+        }}
+        ref={zoomRef}
       >
-        {isValidElement<{ className: string }>(children)
-          ? cloneElement(children, { className: classNames(`${baseClassName}-child`, children?.props?.className) })
-          : children}
-      </TransformComponent>
-    </TransformWrapper>
-  );
-};
+        <TransformComponent
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          wrapperProps={{ ...commonProps, ...props }}
+          wrapperClass={classNames(baseClassName, className, {
+            [`${baseClassName}-zoomed`]: isZoomed,
+          })}
+          contentClass={`${baseClassName}-content`}
+        >
+          {isValidElement<{ className: string; ref: React.Ref<HTMLElement> }>(children)
+            ? cloneElement(children, {
+                className: classNames(`${baseClassName}-child`, children?.props?.className),
+                ref,
```

Comment:
I guess passing this ref into the child is sort of unnecessary, might make sense to do this sort of pattern on the `zoomRef` above? allows you to have shared access from the parent and still use it as the ref on the wrapping zoom component (though not sure how often you'll do that)

```
const zoomRef = useRef(null);
useImperativeHandle(ref, () => zoomRef.current, [])
```
https://stackoverflow.com/a/77055452
https://react.dev/reference/react/useImperativeHandle

Reply:
the `zoomRef` unfortunately doesn't get set to an element in the DOM so I can't use that to merge them.